### PR TITLE
Only highlight inner area of view with cursorline

### LINKED ISSUE #2912: Restrict cursorline to editing area
The newly merged cursorline currently is drawn across the whole window, including the gutter. I think it would look better if the cursorline is only drawn in the editing area, that's how it's done in VSCode, Vim, etc. We also already can set the background (for the numbers) with `ui.linenr.selected`, so it's a bit redundant.

### PR DESCRIPTION
**Pull request by A-Walrus**
Fixes #2912

### CODE REVIEW

File: helix-term/src/ui/editor.rs [hunk 1]
```diff
@@ -717,9 +717,9 @@ impl EditorView {
 
         for line in view.offset.row..(last_line + 1) {
             let area = Rect::new(
-                view.area.x,
-                view.area.y + (line - view.offset.row) as u16,
-                view.area.width,
+                view.inner_area().x,
+                view.inner_area().y + (line - view.offset.row) as u16,
+                view.inner_area().width,
```
Comment (archseer) [inline helix-term/src/ui/editor.rs:722]: `let inner_area = view.inner_area()` before this line. Otherwise you're calculating it three times.

**Review by archseer — CHANGES_REQUESTED** [review]

**Review by sudormrfbin — APPROVED** [review]

**Comment by archseer** [comment]
I actually like the current behavior though.

**Comment by ChrHorn** [comment]
> I actually like the current behavior though.

With this PR it is still possible to set `ui.linenr.selected` background color same as cursorline to achieve the current behavior (minus the diagnostics and the gap).

**Comment by archseer** [comment]
Yeah it's not the same, it won't highlight all the gutters.

**Comment by ChrHorn** [comment]
> Yeah it's not the same, it won't highlight all the gutters.

I see. What if we just use `ui.linenr(.selected)` for the whole gutter and rename it to `ui.gutter(.selected)` (`ui.gutter` is already used but not documented). Think this looks better in general and would allow both styles for the cursorline.

Could also use a more fine-grained approach where every selected component can be themed, but I'm not sure if this is really needed.

**Comment by archseer** [comment]
`ui.gutter.selected` is a good idea, but I'd also keep `ui.linenr.selected` so that only the line number foreground can be changed.

**Comment by pascalkuthe** [comment]
closing this one out as stale, the entire rendering code has been replaced (and will see further large changes) so it would be easier to start over with this one. Thank ou for contributing!

**Pull request closed without merging**